compiler/p/codegen/PPCSystemLinkage.cpp

```diff
@@ -1441,14 +1441,19 @@ void TR::PPCSystemLinkage::buildDirectCall(TR::Node *callNode,
    int32_t                        refNum = callSymRef->getReferenceNumber();
 
    //This is not JIT pseudo TOC, but jit-module system TOC.
+   //Unfortunately, Ruby JIT is multiplexing gr2(as system module TOC) for JIT-pseudo TOC
+
 
    if(aix_style_linkage)
          {
          //Implies TOC needs to be restored (ie not 32bit ppc-linux-be)
          //This is wrong with regard to where system TOC is restored from, but happens to be right for JIT
          //for the time being.
          if(TR::Compiler->target.cpu.isBigEndian())
-            TR::TreeEvaluator::restoreTOCRegister(callNode, cg(), dependencies);
+#if !(defined(PYTHON) || defined(OMR_RUBY) || defined(JITTEST))
+            TR::TreeEvaluator::restoreTOCRegister(callNode, cg(), dependencies)
+#endif
+            ;
```

Comment:
Interesting use of a semicolon statement. You may want to consider introducing brackets for the if clause (create a scope for the ifdef) and moving the semicolon to line 1454. This should improve readability. It also guards against someone in the future accidentally adding another statement in the ifdef not realizing what is happening here.